actions/tag/calculate-semver/entrypoint/main_test.go

```diff
@@ -200,11 +204,39 @@ func TestEntrypoint(t *testing.T) {
 					fmt.Fprintln(w, `{ "message": "Malformed JSON }`)
 
 				default:
-					t.Fatal(fmt.Sprintf("unknown request: %s", dump))
+					t.Fatalf("unknown request: %s", dump)
 				}
 			}))
+
+			file, err := os.CreateTemp("", "output")
+			Expect(err).NotTo(HaveOccurred())
+			githubSetOutputFile = file.Name()
+			Expect(file.Close()).To(Succeed())
+
+			file, err = os.CreateTemp("", "state")
+			Expect(err).NotTo(HaveOccurred())
+			githubSaveStateFile = file.Name()
+			Expect(file.Close()).To(Succeed())
 		})
 
+		it.After(func() {
+			if githubSetOutputFile != "" {
+				Expect(os.Remove(githubSetOutputFile)).To(Succeed())
+				githubSetOutputFile = ""
+			}
+			if githubSaveStateFile != "" {
+				Expect(os.Remove(githubSaveStateFile)).To(Succeed())
+				githubSaveStateFile = ""
+			}
+		})
+
+		outputContains := func(elements ...string) {
```

Comment:
Have you tested what happens when this fails? Specifically, does the failure message get associated with the correct line?

I have vague memories of using helper functions like this resulting in incorrect line info, but that might have been specific to `ginkgo`, rather than `go test` + `spec` library.

Reply:
The line reported will be in the helper function. Actually, there is [`ExpectWithOffset`](https://onsi.github.io/gomega/#making-assertions-in-helper-functions) for this use-cases, but the offset seems to be ignored (maybe only works with `ginkgo`?)

Example output:

```
--- FAIL: TestEntrypoint (0.69s)
    --- FAIL: TestEntrypoint/calculate-semver (0.22s)
        --- FAIL: TestEntrypoint/calculate-semver/all_PRs_since_the_last_release_have_semver:patch/increments_the_patch_from_the_previous_version (0.01s)
            main_test.go:231: 
                Expected
                    <[]string | len:2, cap:2>: ["tag=1.2.4", ""]
                to contain elements
                    <[]string | len:1, cap:1>: ["tag=1.2.5"]
                the missing elements were
                    <[]string | len:1, cap:1>: ["tag=1.2.5"]
FAIL
FAIL	github.com/paketo-buildpacks/github-config/actions/tag/calculate-semver-tag/entrypoint	0.696s
FAIL
```

While the line is not to helpful, the test name is printed, which should be sufficient to find the relevant part, WDYT?
Alternatively, I could make the function return the output array and move the matcher call to to individual tests, if you prefer.

Reply:
I think the `ExpectWithOffset` is what I was thinking of, and it doesn't surprise me that it's only required/only works when using Ginkgo.

I think this is fine - you can see which test is failing and what the expected/desired cases are.